be1-go/hub/standard_hub/hub_state/channels_by_id.go

```diff
@@ -0,0 +1,44 @@
+package hub_state
+
+import (
+	"maps"
+	"popstellar/channel"
+	"sync"
+)
+
+// ChannelsById provides a thread-safe structure that stores channel ids with their corresponding channels
+type ChannelsById struct {
+	sync.RWMutex
+	table map[string]channel.Channel
+}
```

Comment:
This struct looks very much like a thread-safe map.
Have you considered creating a generic instead ?

```suggestion
type ThreadSafeMap[K comparable, V any] struct {
	sync.RWMutex
	m map[K]V
}

type ChannelsById ThreadSafeMap[string, channel.Channel]
```